File: python/greatest_product.py
def greatest_product(n):
    max_product = 0

    for i in range(len(n) - 4):
        chunk = n[i:i+5]
        product = 1
        for char in chunk:
            product *= int(char)
        if product > max_product:
            max_product = product

    return max_product

File: javascript/greatestProduct.js
function greatestProduct(n) {
    let maxProduct = 0;

    for (let i = 0; i <= n.length - 5; i++) {
        const chunk = n.slice(i, i + 5);
        const product = chunk.split('').reduce((prod, char) => prod * parseInt(char), 1);
        if (product > maxProduct) {
            maxProduct = product;
        }
    }

    return maxProduct;
}